viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -333,97 +371,82 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in ARGB32 format
     auto painted_surface = painted_slices->surface.get();
+    auto image_format = cairo_image_surface_get_format(painted_surface);
+    if (image_format != cartographer::io::kCairoFormat) {
+        std::string error_log =
+            "Error cairo surface in wrong format, expected Cairo_Format_ARGB32";
+        LOG(ERROR) << error_log;
```

Comment:
```suggestion
        LOG(ERROR) << error_log << " got " <<  image_format;
```

Reply:
This is non-trivial as it is a custom type that when cast to string just appears as 0,1,2,3,4... In the future we could add a utility function to convert this into a usage string but since this can not be affected by us or our users and only by fundamental changes to cartographer we will not pursue this at this point.